### PR TITLE
add optional flag for full block responses

### PR DESCRIPTION
**Pull request by koko1123**
Add flag in block and block header requests to toggle between full and light response types

### CODE REVIEW

File: protobuf/flow/access/access.proto [hunk 1]
```diff
@@ -128,32 +128,39 @@ message PingResponse {}
...
 
 message GetBlockHeaderByHeightRequest {
   uint64 height = 1;
+  optional bool full_block_header = 2;
 }
 
 message BlockHeaderResponse {
   entities.BlockHeader block = 1;
+  optional bool full_block_header = 2;
 }
```
Comment (smnzhu) [inline protobuf/flow/access/access.proto:147]: Do we actually need this for block header? I thought the size increase was only relevant for the full block?

File: protobuf/flow/access/access.proto [hunk 2]
```diff
@@ -128,32 +128,39 @@ message PingResponse {}
...
 
 message GetBlockHeaderByHeightRequest {
   uint64 height = 1;
+  optional bool full_block_header = 2;
 }
 
 message BlockHeaderResponse {
   entities.BlockHeader block = 1;
+  optional bool full_block_header = 2;
```
Comment (smnzhu) [inline protobuf/flow/access/access.proto:146]: probably not necessary for the response

File: protobuf/flow/access/access.proto (hunk 1 above)
Comment (koko1123) [reply protobuf/flow/access/access.proto:147]: I did it to cut down on `ParentVoterIDs` and `ParentVoterSigData` that used to just be a composite object

Comment (koko1123) [reply protobuf/flow/access/access.proto]: actually nver mind, you're right, even the signer ID size doesn't baloon too much, and we don't include the block heder in the light version of the `block` response anyway

**Review by smnzhu — APPROVED** [review]

**Pull request merged**